### DISCUSSION TITLE
Trigger full rebuild of package before installation

### DISCUSSION THREAD

**Discussion by tibortakacs** [tox-dev/tox#3028] [Q&A]
I have a package with a `key.json` data file in it. `setup.cfg`:
```
[options.package_data]
enigma = key.json
```
When I build it, it properly copies the data file into the package, and installs it into the relevant tox environment.

However, if I modify this content to `enigma = key_missing.json`, I would expect failure in tests due to the missing data file. Then, I renamed `key.json` to `key_missing.json`, and now both files are part of the package.

I think this is not a `tox` issue directly, but it would be great I could ensure that my package is completely re-built. What are you thoughts on that?

**Comment by gaborbernat** [comment]
We do ensure this, if anything is a setuptools bug. I suggest using a more modern build backend, like hatchling.

**Reply by tibortakacs** [reply]
Thank you, make sense. Is there an example how the default build backend can be replaced in tox?

**Reply by gaborbernat** [reply]
This is not tox specific. tox follows the Python specs. https://tox.wiki/en/latest/user_guide.html#packaging

**Reply by tibortakacs** [reply]
Thank you, I am going to read it through.
